src/HIE/Bios/Environment.hs

```diff
@@ -223,7 +225,42 @@ disableOptimisation df = updOptLevel 0 df
 getTargetsFromGhciScript :: FilePath -> IO [String]
 getTargetsFromGhciScript script = do
   contents <- lines <$> readFile script
-  return
-    $ concatMap (tail {- first element is ":add" which we want to remove -}
-                      . words)
-    $ filter (":add" `isPrefixOf`) contents
+  let parseGhciLine = concatMap fst . filter (null . snd) . readP_to_S parser
+  return $ concatMap parseGhciLine contents
+
+-- |This parser aims to parse targets and double-quoted filepaths that are separated by spaces
+-- and prefixed with the literal ":add"
+--
+-- >>> filter (null . snd) $ readP_to_S parser ":add Lib Lib2"
+-- [(["Lib","Lib2"],"")]
+--
+-- >>> filter (null . snd) $ readP_to_S parser ":add Lib Lib2 \"Test Example.hs\""
+-- [(["Lib","Lib2","Test Example.hs"],"")]
+--
+-- >>> filter (null . snd) $ readP_to_S parser ":add Lib Lib2 \"Test Exa\\\"mple.hs\""
+-- [(["Lib","Lib2","Test Exa\"mple.hs"],"")]
+parser :: ReadP [String]
+parser = do
+  _ <- string ":add "
+  scriptword `sepBy` space1
+
+space1 :: ReadP [Char]
+space1 = many1 (char ' ')
+
+scriptword :: ReadP String
+scriptword = quoted <++ value
+
+-- | A balanced double-quoted string
+quoted :: ReadP String
+quoted = do
+    _ <- char '"'
+    manyTill (escaped '"' <|> anyToken) $ char '"'
+
+escaped :: Char -> ReadP Char
+escaped c = c <$ string ("\\" <> [c])
+
+value :: ReadP String
+value = many1 (satisfy (not . isSpace))
+
+anyToken :: ReadP Char
+anyToken = satisfy $ const True
```

Comment:
Pretty sure that not all tokens are valid here.

Reply:
True, only those that are valid in module names. But I dont think we want to validate these module names here and when attempting to load, we crash anyways.
But we can change it.